client/clientfactory.go

```diff
@@ -358,3 +374,8 @@ func (cf *rpcClientFactory) newFrontendGRPCClient(hostAddress string) (frontend.
 		apiv1.NewVisibilityAPIYARPCClient(config),
 	), nil
 }
+
+func isGRPCOutbound(config transport.ClientConfig) bool {
+	namer, ok := config.GetUnaryOutbound().(transport.Namer)
```

Comment:
should we panic/log.fatal here if `ok==false`?

Reply:
Unless old version of yarpc is used, this will not happen. But I guess explicit early panic is better. Updated.